sdk/src/derivation_path.rs

```diff
@@ -142,16 +144,26 @@ impl DerivationPath {
                     "invalid query string, extra fields not supported".to_string(),
                 ));
             }
-            let key = query.get("key");
-            if key.is_none() {
+            let key = query.get(&QueryKey::Key.to_string());
+            if let Some(key) = key {
```

Comment:
wdyt about attempting to query both `key` and `full-path`, then implementing the rest of this logic in a match statement?

Motivation being to throw an error should _both_ queries be specified

Reply:
It actually already does throw if both are specified (and there's a test case covering) due to this check:
https://github.com/solana-labs/solana/blob/1fe131a5bb4eca7a7a0fb83dbeaacdf1f640ca60/sdk/src/derivation_path.rs#L134

...but I'll take a closer look at whether match might be more readable here anyway

Reply:
It's more lines, even with the comment removed, and I honestly don't think it's more readable. I guess just a stylistic decision between early returns and match. I prefer the former in this case.

```diff
diff --git a/sdk/src/derivation_path.rs b/sdk/src/derivation_path.rs
index 4ed308174..069443ccd 100644
--- a/sdk/src/derivation_path.rs
+++ b/sdk/src/derivation_path.rs
@@ -153,25 +153,26 @@ impl DerivationPath {
                 ));
             }
             let key = query.get(QueryKey::Key.as_ref());
-            if let Some(key) = key {
-                // Use from_key_str instead of TryInto here to make it more explicit that this
-                // generates a Solana bip44 DerivationPath
-                return Self::from_key_str(key).map(Some);
-            }
-            if key_only {
-                return Err(DerivationPathError::InvalidDerivationPath(format!(
-                    "invalid query string `{}`, only `key` supported",
-                    query_str,
-                )));
-            }
             let full_path = query.get(QueryKey::FullPath.as_ref());
-            if let Some(full_path) = full_path {
-                return Self::from_absolute_path_str(full_path).map(Some);
+
+            match (key, full_path) {
+                (Some(key), None) => Self::from_key_str(key).map(Some),
+                (None, Some(full_path)) => {
+                    if key_only {
+                        Err(DerivationPathError::InvalidDerivationPath(format!(
+                            "invalid query string `{}`, only `key` supported",
+                            query_str,
+                        )))
+                    } else {
+                        Self::from_absolute_path_str(full_path).map(Some)
+                    }
+                }
+                (None, None) => Err(DerivationPathError::InvalidDerivationPath(format!(
+                    "invalid query string `{}`, only `key` and `full-path` supported",
+                    query_str,
+                ))),
+                _ => unreachable!(),
             }
-            Err(DerivationPathError::InvalidDerivationPath(format!(
-                "invalid query string `{}`, only `key` and `full-path` supported",
-                query_str,
-            )))
         } else {
             Ok(None)
         }
```

Reply:
Could fold the `if key_only` into a match guard and eliminate two lines :wink: 

I guess I find it more readable due to the error and success conditions being resolved in the same statement, rather than via early-returns. I won't block on this either way